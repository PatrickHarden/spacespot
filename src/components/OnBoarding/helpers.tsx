import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
