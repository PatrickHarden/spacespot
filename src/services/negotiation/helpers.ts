import { get } from 'lodash'
import {
  NegotiationTerms,
  NegotiationTerm,
  NegotiationTermFields,
  NegotiationTermFieldsIds,
} from './types'

export const getNegotiationTermArray = (
  terms: NegotiationTerms,
  ids: NegotiationTermFieldsIds,
) => {
  const values: Array<NegotiationTerm> = Object.keys(terms).map(key => {
    return {
      termId: ids[key],
      value: get(terms, key),
    }
  })
  return values
}

const decapitalize = (title: string) => title[0].toLowerCase() + title.slice(1)

export const getFields = (
  fields: NegotiationTermFields,
  negTerms: Array<NegotiationTerm>,
): NegotiationTerms => {
  const initialTerms = {} as NegotiationTerms
  if (!negTerms || !fields) return initialTerms
  return (negTerms.reduce(
    (acc, term) => ({
      ...acc,
      [decapitalize(fields[term.termId])]: term.value,
    }),
    initialTerms,
  ) as unknown) as NegotiationTerms
}
