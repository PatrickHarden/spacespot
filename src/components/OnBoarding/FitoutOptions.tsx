import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'

import actions from 'services/onboarding/actions'
import selectors from 'services/onboarding/selectors'
import { SpaceData } from 'services/onboarding/types'
import { FitoutOptionProps } from 'services/negotiation/types'

import { useIntl } from 'react-intl'
import PlusIcon from 'components/icons/Plus'
import Heading3 from 'components/common/Heading3'
import { getRegionCurrencyDesc } from 'services/global/region'
import OutlinedButton from 'components/common/OutlinedButton'
import Clear from '@material-ui/icons/Clear'

import FitoutForm from './FitoutForm'
import HTMLContent from 'components/common/HTMLContent'

const Container = styled.div``

const FitoutAllContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
`

const FitoutContainer = styled.div`
  margin: 20px 0 0 0;
  width: 100%;
  padding: 20px;
  box-sizing: border-box;
  border: 1px solid #dddddd;
  background-color: #f4f4f4;
  position: relative;
`

const FitoutDescription = styled.div`
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
`
const ClearIcon = styled(Clear)`
  color: #000000;
  font-size: 15px;
  font-weight: 500;
  position: absolute;
  top: 10px;
  right: 10px;
  cursor: pointer;
  background-color: trasparent;
`

const FitoutOptionTitle = styled.div`
  color: #000000;
  font-weight: 500;
`
const FitoutOptionDescription = styled.div`
  margin: 10px 0;
`
const FitoutOptionPrice = styled.div`
  color: #000000;
  font-weight: 500;
`
const AddFitout = styled(OutlinedButton)`
  margin-top: 20px;
  box-sizing: border-box;
  svg {
    margin: 0px 5px -3px 0px;
  }
`

const Plus = styled(PlusIcon)`
  margin: -2px 5px;
`
const FitoutOptions = (props: { space: SpaceData }) => {
  const dispatch = useDispatch()
  const state = useSelector(selectors.onboardingState)
  const country = state ? state.country : ''
  const { space } = props
  const { fitout } = space
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const [show, setShow] = useState(false)

  const deleteFitout = (index: number) =>
    dispatch(
      actions.setSpaceFitout(
        space.id,
        fitout.filter(
          (_option: FitoutOptionProps, indexFilter: number) =>
            index !== indexFilter,
        ),
      ),
    )

  const currency = getRegionCurrencyDesc(country)
  const numberWithCommas = (x: number) => {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  }
  const saveFitout = (newFitout: FitoutOptionProps) => {
    setShow(false)
    dispatch(actions.setSpaceFitout(space.id, [...fitout, newFitout]))
  }

  return (
    <Container>
      <Heading3>{t('ONBOARDING_FITOUT_OPTIONS_TITLE')}</Heading3>
      <FitoutDescription>
        {t('ONBOARDING_FITOUT_OPTIONS_DESCRIPTION')}
      </FitoutDescription>
      <FitoutAllContainer>
        {fitout &&
          fitout.map((option: FitoutOptionProps, index: number) => (
            <FitoutContainer key={option.name}>
              <ClearIcon
                fill="#000000"
                strokeWidth="1"
                onClick={() => deleteFitout(index)}
              />
              <FitoutOptionTitle>{option.name}</FitoutOptionTitle>
              <FitoutOptionDescription>
                <HTMLContent html={option.description} />
              </FitoutOptionDescription>
              <FitoutOptionPrice>
                {`${numberWithCommas(option.amount)}${currency}`}
              </FitoutOptionPrice>
            </FitoutContainer>
          ))}
      </FitoutAllContainer>
      {show ? (
        <FitoutForm setShow={() => setShow(false)} saveFitout={saveFitout} />
      ) : (
        <AddFitout onClick={() => setShow(true)}>
          <Plus size="1.1em" />
          {t('ONBOARDING_FITOUT_OPTIONS_ADD')}
        </AddFitout>
      )}
    </Container>
  )
}
export default FitoutOptions
