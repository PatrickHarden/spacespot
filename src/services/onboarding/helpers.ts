import { get, cloneDeep } from 'lodash'
import { Space } from 'services/space/types'
import spaceSelectors from 'services/space/selectors'
import {
  AmenityOptions,
  OnBoardingState,
  AvailabilityFlexPriceInterval,
  AvailabilityFlex,
} from 'services/onboarding/types'

export const aspects2Amenities = (
  amenities: AmenityOptions,
  aspects: Array<string>,
): AmenityOptions => {
  const newAmenities = cloneDeep(amenities) || {}
  if (!aspects) {
    return newAmenities
  }
  aspects.forEach(key => {
    if (newAmenities[key]) {
      newAmenities[key].checked = true
    } else {
      newAmenities[key] = {
        checked: true,
        desc: key,
      }
    }
  })
  return newAmenities
}

export const loadOnboardingSpace = (
  defaultAmenities: AmenityOptions | undefined,
  space: Space,
  isBuilding: boolean,
): OnBoardingState => {
  const building = spaceSelectors.translateBuilding(space, isBuilding)
  return {
    buildingId: building.buildingId,
    isNew: false,
    images: building.images || [],
    docs: [],
    name: building.name,
    description: building.description,
    address: building.address,
    city: building.city,
    postCode: building.postCode,
    country: building.country,
    location: building.location || '',
    latLng: building.latLng,
    district: building.district,
    subdistrict: building.subdistrict,
    amenities: aspects2Amenities(
      defaultAmenities || {},
      get(space, 'Common.Aspects', []),
    ),
    spaces: isBuilding ? {} : spaceSelectors.translateSpace(space),
    spaceCount: 0,
    uploadedImages: building.uploadedImages,
    uploadedDocs: building.uploadedDocs,
    isPublishing: false,
    defaultAmenities: defaultAmenities,
  }
}

const defaultDate = new Date()
defaultDate.setDate(new Date().getDate() + 1)

export const getDefaultAvailability = (): AvailabilityFlex => ({
  desks: 0,
  availableFrom: defaultDate,
  minLease: 0,
  price: 0,
  currencyCode: 'NOK',
  frequency: AvailabilityFlexPriceInterval.Monthly,
})
