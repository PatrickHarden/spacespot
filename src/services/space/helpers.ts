import { get, chain } from 'lodash'
import { Space } from './types'
import {
  CommonChargeType,
  CommonImage,
  ImageResource,
  CommonPostalAddress,
  CommonActualAddress,
} from 'services/space/PropertyListingschema'
import { FLEX_TYPE_CONSTANT, FIXED_TYPE_CONSTANT } from '../onboarding/utils'
import { routingLang } from 'intl'

import CONFIG from 'config'

export const generateGallery = (item: any) => {
  const [url] = item['Common.ImageResources'].filter(
    (photo: any) => photo['Common.Breakpoint'] === 'small',
  )
  const alt = item['Common.ImageCaption']
  return {
    url: url['Source.Uri'],
    alt,
  }
}

type Charges = {
  rent: number
  serviceCharges: number
  additionalCosts: number
  currencyCode: string
  total: number
}

export const getCharges = (space: Space) => {
  let charges: Charges = {
    rent: 0,
    serviceCharges: 0,
    additionalCosts: 0,
    currencyCode: 'EUR',
    total: 0,
  }
  const ccharges: Array<CommonChargeType> = get(space, 'Common.Charges', [])
  charges = ccharges.reduce((acc, charge) => {
    const kind = charge['Common.ChargeKind']
    const value = charge['Common.Amount'] || 0
    const currency = charge['Common.CurrencyCode']
    if (currency) {
      acc.currencyCode = currency
    }
    switch (kind) {
      case 'Rent':
        acc.rent = value
        break
      case 'ServiceCharge':
        acc.serviceCharges += value
        break
      default:
        acc.additionalCosts += value
    }
    acc.total += value
    return acc
  }, charges)
  return charges
}

export const getImages = (space: Space) => {
  const photos = get(space, 'Common.Photos')
  const urls = photos
    ? photos.map((item: CommonImage) => {
      const [original] = item['Common.ImageResources'].filter(
        (photo: ImageResource) => photo['Common.Breakpoint'] === 'original',
      )
      const [small] = item['Common.ImageResources'].filter(
        (photo: ImageResource) => photo['Common.Breakpoint'] === 'small',
      )
      return {
        original:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
        thumbnail:
          small['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + small['Common.Resource.Uri'],
      }
    })
    : []
  return urls
}

export const getFloorPlans = (space: Space) => {
  const floorPlans = get(space, 'Common.FloorPlans')
  const urls = floorPlans
    ? floorPlans.map((item: CommonImage) => {
      const original = item['Common.ImageResources'][0]
      return {
        original:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
        thumbnail:
          original['Source.Uri'] ||
          CONFIG.GL_PHOTO_HOST + original['Common.Resource.Uri'],
      }
    })
    : []
  return urls
}

export const getPhoto = (space: Space): string => {
  const photo = chain(space)
    .get('Common.Photos')
    .head()
    .get('Common.ImageResources')
    .filter((img: ImageResource) => img['Common.Breakpoint'] === 'small')
    .head()
    .get('Source.Uri')
    .value()
  return photo || ''
}

export const getType = (space: Space) =>
  get(space, 'Common.UsageType') === 'Office' //FIXME when mapping changes
    ? FIXED_TYPE_CONSTANT
    : FLEX_TYPE_CONSTANT

export const isFixed = (space: Space) => getType(space) === FIXED_TYPE_CONSTANT

export const getSize = (space: Space) => {
  if (isFixed(space)) {
    return get(space, "['Common.MinimumSize'][0]['Common.Size']", 0)
  }
  return 0 // FIXME
}

/**
 * Size units (i.e: sqm)
 */
export const getSizeUnits = (space: Space) => {
  if (isFixed(space)) {
    return get(space, "['Common.MinimumSize'][0]['Common.Units']", 0)
  }
  return 0 // FIXME
}

/**
 * Calculate desks from size for fixed spaces
 *
 * desks = size in sqm / 15
 *
 * - For flex/coworking returns empty string
 * - If units != sqm or sqft, returns size with units
 *
 */
export const getDesks = (space: Space) => {
  if (isFixed(space)) {
    const sz: number = getSize(space)
    const units = getSizeUnits(space)
    if (units === 'sqm' || units === 'm2') {
      return (sz / 15).toFixed()
    } else if (units === 'sqft') {
      return ((sz * 0.092903) / 15).toFixed()
    } else {
      // unknown units
      return `${sz} ${units}`
    }
  }
  return ''
}

/**
 * Get common area (shared) size
 *
 * Shared common area = TotalSize - Size
 */
export const getSizeCommon = (space: Space) => {
  if (isFixed(space)) {
    return (
      get(space, "['Common.TotalSize'][0]['Common.Size']", 0) - getSize(space)
    )
  }
  return 0
}

export const getFloor = (space: Space) => get(space, 'Common.FloorNumber')

const getPostalAddress = (space: Space): CommonPostalAddress => {
  const actual: CommonActualAddress = get(space, 'Common.ActualAddress')
  const postal: Array<CommonPostalAddress> = get(
    actual,
    'Common.PostalAddresses',
    [],
  )
  return postal[0]
}

export const getLocation = (space: Space) =>
  get(space, '["Common.LocationDescription"][0]["Common.Text"]')

export const getTitle = (space: Space): string =>
  get(
    space,
    '["Common.Strapline"][0]["Common.Text"]',
    get(getPostalAddress(space), 'Common.Line1', ''),
  )

export const getStreet = (space: Space): string => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Line2', get(address, 'Common.Line1', ''))
}

export const getDistrict = (space: Space) => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Line3', '')
}

export const getSubdistrict = (space: Space) => {
  const address = getPostalAddress(space)
  return {
    id: 0,
    name: get(address, 'Common.Line4', ''),
  }
}

export const getCity = (space: Space): string => {
  const address = getPostalAddress(space)
  return get(address, 'Common.Locallity', '')
}

export const getPK = (space: Space): string =>
  get(space, 'Common.PrimaryKey', '')

export const getSpaceURI = (space: Space): string => {
  const office = routingLang({ nb: 'kontor', default: 'office' })
  const coworking = 'coworking'

  const id = getPK(space)
  const address = getStreet(space).replace(/\//g, '-')
  const district = (getDistrict(space) || 'NA').replace(/\//g, '-')
  const subdistrict = (getSubdistrict(space).name || 'NA').replace(/\//g, '-')
  const city = getCity(space)
  const title = getTitle(space).replace(/\//g, '-')
  const stype = isFixed(space) ? office : coworking
  const uri = `detail/${stype}/${city}/${district}/${subdistrict}/${id}/${title}-${address}`
  const repKeys: { [key: string]: string } = {
    æ: 'ae',
    å: 'aa',
    ø: 'oe',
    Æ: 'Ae',
    Å: 'Aa',
    Ø: 'Oe',
    á: 'a',
    é: 'e',
    í: 'i',
    ó: 'o',
    ú: 'u',
    Á: 'A',
    É: 'E',
    Í: 'I',
    Ó: 'O',
    Ú: 'U',
    '–': '-',
    '|': '-',
    ä: 'a',
    ë: 'e',
    ï: 'i',
    ö: 'o',
    ü: 'u',
    Ä: 'A',
    Ë: 'E',
    Ï: 'I',
    Ö: 'O',
    Ü: 'U',
  }
  return uri
    .replace(/['()&?,. ]/g, '-')
    .replace(/\*/g, '-')
    .replace(/[æåøÆÅØáéíóúÁÉÍÓÚ–|äëïöüÄËÏÖÜ]/g, char => repKeys[char] || '')
    .replace(/-+/g, '-')
}
