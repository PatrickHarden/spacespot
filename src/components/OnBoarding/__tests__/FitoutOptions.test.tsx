import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { createSpace } from '../helpers'

import FitoutOptions from '../FitoutOptions'

jest.mock('react-quill', () => {
  const ComponentToMock = (props: any) => <textarea {...props} />
  return ComponentToMock
})

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  useDispatch: () => {
    return mockDispatch
  },
  useSelector: () => () => '',
}))

it('renders without crashing ', async () => {
  const wrapper1 = mount(<FitoutOptions space={createSpace('FLEX')} />)
  expect(mountToJson(wrapper1)).toMatchSnapshot()
  // open form
  wrapper1.find('button').simulate('click')
  act(() => {
    wrapper1.update()
  })
  // fill form
  wrapper1
    .find('input[type="text"]')
    .simulate('change', { target: { value: 'name1' } })
  wrapper1.find('textarea').simulate('change', { target: { value: 'desc1' } })
  wrapper1
    .find('input[type="number"]')
    .simulate('change', { target: { value: '2' } })
  const buttons = wrapper1.find('button')
  act(() => {
    // save fitout
    buttons.at(1).simulate('click')
    wrapper1.update()
    // cancel fitout
    wrapper1
      .find('button')
      .at(0)
      .simulate('click')
    // remove
    wrapper1.find('svg').simulate('click')
  })
})
