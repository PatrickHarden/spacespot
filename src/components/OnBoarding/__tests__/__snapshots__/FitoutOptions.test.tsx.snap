// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing  1`] = `
<FitoutOptions
  space={
    Object {
      "availabilityFixed": 2020-01-01T00:00:00.000Z,
      "fitout": Array [
        Object {
          "amount": 1000,
          "description": "FitOut1",
          "name": "FitOut1",
        },
      ],
      "fixedDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "floorPlan": Array [],
      "hotDesks": Object {
        "availableFrom": 2020-01-01T00:00:00.000Z,
        "currencyCode": "NOK",
        "desks": 0,
        "frequency": "Monthly",
        "minLease": 0,
        "price": 0,
      },
      "id": "0",
      "images": Array [],
      "months": undefined,
      "servicedOffices": Array [],
      "spaceDescription": "",
      "spaceFloor": undefined,
      "spaceFloored": "",
      "spaceHighlights": "",
      "spaceMatterPort": "",
      "spaceName": "",
      "spaceRent": undefined,
      "spaceServices": undefined,
      "spaceServicesNotNegotiable": false,
      "spaceSize": undefined,
      "spaceSizeCommon": undefined,
      "status": 1,
      "type": "FLEX",
      "use": "",
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-kAzzGY",
            "isStatic": false,
            "lastClassName": "jHprQs",
            "rules": Array [],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-kAzzGY",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-kAzzGY jHprQs"
      >
        <styled.h3>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "klePfV",
                  "rules": Array [
                    "
  color: #50bbd8;
  line-height: 25px;
  font-size: 22px;
  font-weight: 400;
",
                  ],
                },
                "displayName": "styled.h3",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "h3",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <h3
              className="sc-bdVaJa klePfV"
            >
              ONBOARDING_FITOUT_OPTIONS_TITLE
            </h3>
          </StyledComponent>
        </styled.h3>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-kGXeez",
                  "isStatic": false,
                  "lastClassName": "IZhOg",
                  "rules": Array [
                    "
  color: #000000;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-kGXeez",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-kGXeez IZhOg"
            >
              ONBOARDING_FITOUT_OPTIONS_DESCRIPTION
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-chPdSV",
                  "isStatic": false,
                  "lastClassName": "gtWHTz",
                  "rules": Array [
                    "
  display: flex;
  flex-wrap: wrap;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-chPdSV",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-chPdSV gtWHTz"
            >
              <styled.div
                key="FitOut1"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-kgoBCf",
                        "isStatic": false,
                        "lastClassName": "kYFBPM",
                        "rules": Array [
                          "
  margin: 20px 0 0 0;
  width: 100%;
  padding: 20px;
  box-sizing: border-box;
  border: 1px solid #dddddd;
  background-color: #f4f4f4;
  position: relative;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-kgoBCf",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-kgoBCf kYFBPM"
                  >
                    <Styled(ClearIcon)
                      fill="#000000"
                      onClick={[Function]}
                      strokeWidth="1"
                    >
                      <StyledComponent
                        fill="#000000"
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "compare": null,
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-kpOJdX",
                              "isStatic": false,
                              "lastClassName": "jXlxnH",
                              "rules": Array [
                                "
  color: #000000;
  font-size: 15px;
  font-weight: 500;
  position: absolute;
  top: 10px;
  right: 10px;
  cursor: pointer;
  background-color: trasparent;
",
                              ],
                            },
                            "displayName": "Styled(ClearIcon)",
                            "foldedComponentIds": Array [],
                            "muiName": "SvgIcon",
                            "render": [Function],
                            "styledComponentId": "sc-kpOJdX",
                            "target": Object {
                              "$$typeof": Symbol(react.memo),
                              "compare": null,
                              "displayName": "ClearIcon",
                              "muiName": "SvgIcon",
                              "type": Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "render": [Function],
                              },
                            },
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                        strokeWidth="1"
                      >
                        <ForwardRef
                          className="sc-kpOJdX jXlxnH"
                          fill="#000000"
                          onClick={[Function]}
                          strokeWidth="1"
                        >
                          <WithStyles(ForwardRef(SvgIcon))
                            className="sc-kpOJdX jXlxnH"
                            fill="#000000"
                            onClick={[Function]}
                            strokeWidth="1"
                          >
                            <ForwardRef(SvgIcon)
                              className="sc-kpOJdX jXlxnH"
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                              fill="#000000"
                              onClick={[Function]}
                              strokeWidth="1"
                            >
                              <svg
                                aria-hidden="true"
                                className="MuiSvgIcon-root sc-kpOJdX jXlxnH"
                                fill="#000000"
                                focusable="false"
                                onClick={[Function]}
                                role="presentation"
                                strokeWidth="1"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef>
                      </StyledComponent>
                    </Styled(ClearIcon)>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-dxgOiQ",
                              "isStatic": false,
                              "lastClassName": "MRLmF",
                              "rules": Array [
                                "
  color: #000000;
  font-weight: 500;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-dxgOiQ",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-dxgOiQ MRLmF"
                        >
                          FitOut1
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-ckVGcZ",
                              "isStatic": false,
                              "lastClassName": "VtkxC",
                              "rules": Array [
                                "
  margin: 10px 0;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-ckVGcZ",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-ckVGcZ VtkxC"
                        >
                          <HTMLContent
                            html="FitOut1"
                          >
                            <div
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "FitOut1",
                                }
                              }
                            />
                          </HTMLContent>
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-jKJlTe",
                              "isStatic": false,
                              "lastClassName": "gAmNLc",
                              "rules": Array [
                                "
  color: #000000;
  font-weight: 500;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-jKJlTe",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-jKJlTe gAmNLc"
                        >
                          1,000Kr
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <Styled(styled.button)
          onClick={[Function]}
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-eNQAEJ",
                  "isStatic": false,
                  "lastClassName": "haHNjL",
                  "rules": Array [
                    "
  font-size: 14px;
  font-weight: 550;
  border: 1px solid #6cb9d5;
  color: #6cb9d5;
  background-color: #ffffff;
  border-radius: 2px;
  cursor: pointer;
  text-transform: uppercase;
  height: 37px;
  padding: 0 20px;
",
                    "
  margin-top: 20px;
  box-sizing: border-box;
  svg {
    margin: 0px 5px -3px 0px;
  }
",
                  ],
                },
                "displayName": "Styled(styled.button)",
                "foldedComponentIds": Array [
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-eNQAEJ",
                "target": "button",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="sc-bwzfXH sc-eNQAEJ haHNjL"
              onClick={[Function]}
            >
              <Styled(Plus)
                size="1.1em"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-hMqMXs",
                        "isStatic": false,
                        "lastClassName": "kaBfg",
                        "rules": Array [
                          "
  margin: -2px 5px;
",
                        ],
                      },
                      "displayName": "Styled(Plus)",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-hMqMXs",
                      "target": [Function],
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  size="1.1em"
                >
                  <Plus
                    className="sc-hMqMXs kaBfg"
                    size="1.1em"
                  >
                    <svg
                      className="sc-hMqMXs kaBfg"
                      height="1.1em"
                      size="1.1em"
                      viewBox="0 0 18 17"
                      width="1.1em"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                        stroke="#6CB9D5"
                        strokeLinecap="round"
                        strokeWidth={2}
                      >
                        <path
                          d="M9 1.6v13.8M1.64 8.5h14.72"
                        />
                      </g>
                    </svg>
                  </Plus>
                </StyledComponent>
              </Styled(Plus)>
              ONBOARDING_FITOUT_OPTIONS_ADD
            </button>
          </StyledComponent>
        </Styled(styled.button)>
      </div>
    </StyledComponent>
  </styled.div>
</FitoutOptions>
`;
